src/rust/catpowder/linux/mod.rs

```diff
@@ -69,40 +67,12 @@ impl LinuxRuntime {
 }
 
 //======================================================================================================================
-// Imports
+// Trait Implementations
 //======================================================================================================================
 
-impl MemoryRuntime for LinuxRuntime {
-    fn sgaalloc(&self, size: usize) -> Result<demi_sgarray_t, Fail> {
-        // TODO: Allocate an array of buffers if requested size is too large for a single buffer.
-
-        if size == 0 {
-            let cause: String = format!("cannot allocate a zero-sized buffer");
-            error!("sgaalloc(): {}", cause);
-            return Err(Fail::new(libc::EINVAL, &cause));
-        }
-
-        if size > u16::MAX as usize {
-            return Err(Fail::new(libc::EINVAL, "size too large for a single demi_sgaseg_t"));
-        }
-
-        // Always allocate with header space for now even if we do not need it.
-        let buf: DemiBuffer = DemiBuffer::new_with_headroom(size as u16, MAX_HEADER_SIZE as u16);
-
-        // Create a scatter-gather segment to expose the DemiBuffer to the user.
-        let data: *const u8 = buf.as_ptr();
-        let sga_seg: demi_sgaseg_t = demi_sgaseg_t {
-            sgaseg_buf: data as *mut c_void,
-            sgaseg_len: size as u32,
-        };
-
-        // Create and return a new scatter-gather array (which inherits the DemiBuffer's reference).
-        Ok(demi_sgarray_t {
-            sga_buf: buf.into_raw().as_ptr() as *mut c_void,
-            sga_numsegs: 1,
-            sga_segs: [sga_seg],
-            sga_addr: unsafe { mem::zeroed() },
-        })
+impl DemiMemoryAllocator for LinuxRuntime {
+    fn allocate_demi_buffer(&self, size: usize) -> Result<DemiBuffer, Fail> {
```

Comment:
Do we need a size check here like SharedDPDKRuntime? Also at other points where a fn with this name exists?